src/number-formatting/index.ts

```diff
@@ -143,13 +148,31 @@ class NumberFormatting implements NumberFormattingOptions {
     /**
      * Returns the actual number of decimals that a number has.
      * @param number - number or numeric string
+     * @param isNoRounding - oprional or boolean

```

Comment:
Is no rounding doesn’t make sense as a parameter here - get actual decimals should always return the actual decimals..

Reply:
> The problem with the existing code is that it assumes that 8 is always going to be a number of decimals that JavaScript can handle with high precision- but because the number is large, that is not the case.
> 
> A different approach (not sure if it is better though) would be to adjust the 8 constant to toFixed depending on how many characters the whole number has.

I think this is something we can do
but what should be total chars we should check against ?
Do you mean to say something like this ?

`
const maxChars = 16;

function getActualDecimals(number) {
        number = Math.abs(number);
        
        let maxDecimals = maxChars  - Math.floor(number).toString().length;
        maxDecimals = Math.max(0, Math.min(maxDecimals, 8))
        
        return (number - Math.floor(number))
            .toFixed(maxDecimals)
            .substring(2, 10)
            .replace(numberOfZerosRx, '').length;
    }
`